### DISCUSSION TITLE
Using `.pre-commit` for automate linting and formatting for the code

### DISCUSSION THREAD

**Discussion by yezz123** [piccolo-orm/piccolo#343] [Ideas]
as I see all the new projects and packages take `pre-commit` as a good package to automate the linting and formatting for the code.
In my own experience, I use `pre-commit` in all my packages and project I see that reduces some commands and work like `./check.sh` or `./lint.sh` then every shell script is coded under the packages that we use for linting or formatting.
We could use the same configuration that we have in the shell script but under a `.yml` file.

cc @dantownsend  what you think you could check this [authx/.pre-commit-config.yaml](https://github.com/yezz123/authx/blob/main/.pre-commit-config.yaml).

**Comment by dantownsend** [comment]
It's a nice idea. I've heard good things about pre-commit, but haven't used it before. I think it makes sense to add it.

**Reply by yezz123** [reply]
that's great then i will take care of adding it 🚀

**Reply by dantownsend** [reply]
Great, thanks!
